### PR TITLE
PR: Check that we can get symbols in a file not part of a Python module

### LINKED ISSUE #13928: The ouline explorer doesn't populate for some files with latest 4.x
## Problem Description
The outline explorer just doesn't want to populate at all for some files on my system. Moreover, the code style and error analysis do not seem to work for these files too.

Here are some lsp logs that I have created by trying to open a very simple file for which the outline explorer doesn't work.

[transport_python_1716.log](https://github.com/spyder-ide/spyder/files/5330042/transport_python_1716.log)
[server_python_1716.log](https://github.com/spyder-ide/spyder/files/5330043/server_python_1716.log)

The content of the file was simply:

```python
# -*- coding: utf-8 -*-


def test_outline_explorer():
    a = 3
    c = b
    return a
```

## Versions

* Spyder version: 4.2.0.dev0 6e0666389
* Python version: 3.7.6 64-bit
* Qt version: 5.12.5
* PyQt5 version: 5.12.3
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0           :  1.4.0 (OK)
chardet >=2.0.0                :  3.0.4 (OK)
cloudpickle >=0.5.0            :  1.5.0 (OK)
diff_match_patch >=20181111    :  20200713 (OK)
intervaltree                   :  None (OK)
IPython >=4.0                  :  7.16.1 (OK)
jedi =0.17.2                   :  0.17.2 (OK)
nbconvert >=4.0                :  5.6.1 (OK)
numpydoc >=0.6.0               :  1.1.0 (OK)
paramiko >=2.4.0               :  2.7.1 (OK)
parso =0.7.0                   :  0.7.0 (OK)
pexpect >=4.4.0                :  4.8.0 (OK)
pickleshare >=0.4              :  0.7.5 (OK)
psutil >=5.3                   :  5.7.2 (OK)
pygments >=2.0                 :  2.6.1 (OK)
pylint >=1.0                   :  2.5.3 (OK)
pyls >=0.35.0;<1.0.0           :  0.35.1+1.gd81c7ba (OK)
qdarkstyle >=2.8               :  2.8.1 (OK)
qtawesome >=0.5.7              :  0.8.0.dev0 (OK)
qtconsole >=4.7.7              :  4.8.0.dev0 (OK)
qtpy >=1.5.0                   :  1.9.0 (OK)
rtree >=0.8.3                  :  0.9.4 (OK)
sphinx >=0.6.6                 :  3.1.2 (OK)
spyder_kernels >=1.9.4;<1.10.0 :  1.10.0.dev0 (OK)
watchdog                       :  None (OK)
zmq >=17                       :  19.0.1 (OK)

# Optional:
cython >=0.21                  :  0.29.21 (OK)
matplotlib >=2.0.0             :  3.2.2 (OK)
numpy >=1.7                    :  1.19.1 (OK)
pandas >=1.1.1                 :  1.0.5 (NOK)
scipy >=0.17.0                 :  1.5.0 (OK)
sympy >=0.7.3                  :  None (OK)
```

### PR DESCRIPTION
**Pull request by ccordoba12**
This was an important regression after moving the Outline to use the LSP symbols information.

Depends on palantir/python-language-server#882
Fixes #13928.

### CODE REVIEW

**Comment by ccordoba12** [comment]
@jnsebgosselin, thanks for the reproducible examples you posted on issue #13928. They helped me to understand the problem and come up with a solution.

Could you test if this PR fixes the problem on Windows? (I added a test but it fails on that OS).

**Comment by jnsebgosselin** [comment]
@ccordoba12 This is great. I'm going to test it today, as soon as I have some time to do it.

**Comment by jnsebgosselin** [comment]
@ccordoba12 I've tested it and I am still able to reproduce the issue. I still need to add a `__init__.py` file for the outline explorer to work.

* Spyder version: 4.2.0.dev0 3bcc6f10c
* Python version: 3.7.6 64-bit
* Qt version: 5.12.5
* PyQt5 version: 5.12.3
* Operating System: Windows 10

image

**Comment by ccordoba12** [comment]
@jnsebgosselin, could you test again? I think I fixed the problem this time.

**Comment by jnsebgosselin** [comment]
> 
> 
> @jnsebgosselin, could you test again? I think I fixed the problem this time.

No, I am still experiencing the issue sorry. Would it help to have the lsp logs?

File: external-deps/python-language-server/pyls/plugins/symbols.py [hunk 1]
```diff
@@ -15,7 +17,13 @@ def pyls_document_symbols(config, document):
     symbols_settings = config.plugin_settings('jedi_symbols')
     all_scopes = symbols_settings.get('all_scopes', True)
     add_import_symbols = symbols_settings.get('include_import_symbols', True)
-    definitions = document.jedi_names(all_scopes=all_scopes)
+
+    use_document_path = False
+    document_dir = os.path.normpath(os.path.dirname(document.path))
```
Comment (ccordoba12) [inline external-deps/python-language-server/pyls/plugins/symbols.py:22]: Please change normpath for abspath here, to see if that helps.

File: external-deps/python-language-server/pyls/workspace.py [hunk 2]
```diff
@@ -263,7 +263,7 @@ def jedi_script(self, position=None, use_document_path=False):
 
         # Extend sys_path with document's path if requested
         if use_document_path:
-            sys_path += [os.path.dirname(self.path)]
+            sys_path += [os.path.normpath(os.path.dirname(self.path))]
```
Comment (ccordoba12) [inline external-deps/python-language-server/pyls/workspace.py:266]: And here too.

**Review by ccordoba12 — COMMENTED** [review]
I can't reproduce your exact conditions because I don't have a D drive on my virtual machne, but  my comments below should help.

**Comment by jnsebgosselin** [comment]
Ok thank you, I will look into this.

**Comment by jnsebgosselin** [comment]
@ccordoba12 how can I make my print to show somewhere in these two files?

**Comment by jnsebgosselin** [comment]
Substituing `os.path.normpath` by `os.path.abspath ` doesn't change anything.

For the record, it doesn't work also if the project is on the C: drive.

**Comment by jnsebgosselin** [comment]
I honestly don't understand how that pyls stuff work at all.

For instance, If I do:

```
@hookimpl
def pyls_document_symbols(config, document):
    return  
```
in the file `.spyder\external-deps\python-language-server\pyls\plugins\symbols.py` does not seem to have any impact on how the outline explorer work, how can that be?

**Comment by jnsebgosselin** [comment]
OK, I got it! My previous testing on this PR were wrong @ccordoba12. 

I was using the `python-language-server` on the develop branch. In order to do that, I added that git repository to my PYTHONPATH in Windows and for some reason, this prevented the overrides done by Spyder to the `pyls` to be applied.

Now that I have removed the `python-language-server` git repo from my PYTHONPATH, this PR indeed fixes the issue as expected.

image

**Comment by ccordoba12** [comment]
Great! Thanks @jnsebgosselin for your help!

**Pull request merged**